src/watson/backends.py

```diff
@@ -21,16 +21,25 @@ def regex_from_word(word):
 
 
 RE_SPACE = re.compile(r"[\s]+", re.UNICODE)
-RE_NON_WORD = re.compile(r"[^ \w\-\.']", re.UNICODE)
 
+# (below was researched and tested on PostgreSQL 9.5)
+# the only chars that to_tsquery does not really like are "! & : ( ) |"
+# "&" does not harm us as it converts to AND which we do anyway
+# "|" is syntactically correct, but performs OR lookup which may not be
+# what the user expects, so we remove it
+RE_NON_WORD = re.compile(r'[!:"(|)]', re.UNICODE)
 
 def escape_query(text):
+    """normalizes the query text to a format that can be consumed
+    by the backend database
+    """
     text = force_text(text)
     text = RE_SPACE.sub(" ", text)  # Standardize spacing.
-    text = RE_NON_WORD.sub("", text)  # Remove non-word characters.
+    text = RE_NON_WORD.sub("&", text)  # Replace harmful characters with logical "AND"
```

Comment:
I think it's best to replace harmful characters with " ", rather than " & ", since then the different backends can re-combine the works using their own version of the & operator. & isn't a logical AND in all search backends.